update steps to deploy using the ops-dashboard (#2707)
DFE-Digital/apply-for-teacher-training: docs/deployment.md

To speed up the release pipeline, logic has been introduced into the pipeline code to only run the full ARM template deployment only if the following conditions are met.
- Any of the pipeline variable groups associated with the deployment have changed since the last successful deployment
- Any of the following files have changed in the commit associated with the build:
  - azure/pipelines/build.yml
  - azure/pipelines/release.yml
  - azure/pipelines/templates/deploy.yml
  - azure/template.json
  - azure/containers.json
If none of the above conditions are met the pipeline will simply load the new image into the container staging slot ready for swapping.

## 1. Check what you're deploying

Go to [the Apply Ops Dashboard](https://apply-ops-dashboard.azurewebsites.net) and find the commit you want to deploy 
and check the diff on GitHub to see if there's anything risky.

You also have to make sure that you're deploying only work that is safe to deploy. It should be either behind a feature flag or product reviewed.

## 2. Deploy to staging

> Click on the `Deploy` button next to the commit sha under staging section in the dashboard.
> You might be prompted to login to Azure AD and grant permissions for the deployment to being, you won't be prompted to login to Azure again if an active session exists.
> Once the deployment beings this will automatically post the list of PRs being deployed to the `#twd_apply` slack channel.

Reviewer: Can we include both methods for now?

E.g.
```suggestion
### :new: Using the deploy button 

Click on the `Deploy` button next to the commit SHA under Staging section in the dashboard.

You might be prompted to login to Azure AD and grant permissions for the deployment. You won't be prompted to login to Azure again if an active session exists.

Once the deployment begins, this will automatically post the list of PRs being deployed to the `#twd_apply` slack channel.

### Manually

Summarise what you're deploying and tell the team in Slack on the `#twd_apply` channel. Use `:ship_it_parrot:` as required.

1. Load the [apply-for-teacher-training-releases](https://dfe-ssp.visualstudio.com/Become-A-Teacher/_build?definitionId=325&_a=summary) page in Azure DevOps.
1. Click the blue "Run pipeline" button (sometimes it says "Queue") at the top right of the page which will open the run pipeline menu.
1. Ensure the branch is set to "master".
1. Specify the commit
1. Under the Variables section, make sure only `deploy_staging` is set to true (this should be the default)
1. Click the Run button to start the deployment.
```

> Click on the `Deploy` button next to the commit sha under staging section in the dashboard.
> You might be prompted to login to Azure AD and grant permissions for the deployment to being, you won't be prompted to login to Azure again if an active session exists.
> Once the deployment beings this will automatically post the list of PRs being deployed to the `#twd_apply` slack channel.

Reviewer: ```suggestion
Click on the `Deploy` button next to the commit SHA under Staging section in the dashboard.

You might be prompted to login to Azure AD and grant permissions for the deployment. You won't be prompted to login to Azure again if an active session exists.

Once the deployment begins, this will automatically post the list of PRs being deployed to the `#twd_apply` slack channel.
```

## 3. Check if staging is deployed successfully

Check the `#twd_apply_tech` channel in Slack for runtime errors from
Sentry or the smoke tests.

## 4. Deploy to production and sandbox

> Click on the `Deploy` button next to the commit sha under the production section of the dashboard.

Reviewer: See above about including both for now.

> Click on the `Deploy` button next to the commit sha under the production section of the dashboard.

Reviewer: ```suggestion
Click on the `Deploy` button next to the commit SHA under the Production section of the dashboard.
```

back is to run a full redeploy._

Because we operate blue/green deployments, the previous version of the app is

> always available in the staging slot. To roll back to it, use tge [swap-slots pipeline](swap-slots-pipeline.md).

Author: ```suggestion
always available in the staging slot. To roll back to it, use the [swap-slots pipeline](swap-slots-pipeline.md).
```
